### ISSUE TITLE
Dark mode for all systems

### ISSUE THREAD

**Issue by akaessens** [jiahaog/nativefier#968]
**Problem statement**
MacOS provides a system interface for using the dark mode. On other platforms a simple solution to enable dark mode is not available. 
[background-color] is not applicable to choose a theme. 
[inject] cannot change the electron theme (as far as i tested)

- [x] I looked at `nativefier --help` and https://github.com/jiahaog/nativefier/blob/master/docs/api.md , no existing option fits my needs.
- [x] I searched existing open & closed issues. Yes, my issue is new.

**Suggested solution**
A [nativeTheme] option to write the eletron nativeTheme (system/dark/light)
[Electron nativeTheme API](https://www.electronjs.org/docs/api/native-theme#nativethemethemesource)

**Alternative solutions**
A simple JS injection script to toggle the theme 
 
**Context**
 - Nativefier: (8.0.7)
 - Node.js: (14.1.0)
 - Npm: (6.14.4)
 - OS: (Manjaro Linux)

**Comment by akaessens** [comment]
Maybe I misunderstood the nativeTheme API, it does not change the webpage itself to a dark theme.
Electron should provide chrome://flags#enable-force-dark as a flag but it doesn't.

**Comment by leuchtrakete666** [comment]
i would love to have an option to build the webapp in dark mode. like the dark reader extension for chrome or ff. i know its possible to do it myself with the injection of a css file. but unfortunately i dont have the skills to do that.

**Comment by being-yash-t** [comment]
> i would love to have an option to build the webapp in dark mode. like the dark reader extension for chrome or ff. i know its possible to do it myself with the injection of a css file. but unfortunately i dont have the skills to do that.

Yes, something like being able to inject a CSS for dark mode can be enough I guess.

**Comment by Fmstrat** [comment]
It would be great if Dark Reader could be integrated.

**Comment by akaessens** [comment]
> > i would love to have an option to build the webapp in dark mode. like the dark reader extension for chrome or ff. i know its possible to do it myself with the injection of a css file. but unfortunately i dont have the skills to do that.
> 
> Yes, something like being able to inject a CSS for dark mode can be enough I guess.

Its very simle with the following steps:
1) find your page on userstyles.org or write your own, example for gitlab: https://userstyles.org/styles/164877/dark-gitlab
2) download JS file via "Install style as userscript"
3) use the inject flag in nativefier

However this always is a custom solution for every site

**Comment by jaewooklee93** [comment]
This feature is quite essential for everyone's eye health.

Since there is no plan to add new feature to support chrome extension, (#548) at least dark reader itself or its analogue should be integrated into nativefier.

**Comment by llabusch93** [comment]
> I've attempted to do this:
> 
> * Download https://unpkg.com/darkreader@4.9.27/darkreader.js
> * Made a `dark.js` with:
> 
> ```js
> DarkReader.enable({
>     brightness: 100,
>     contrast: 90,
>     sepia: 10
> });
> ```
> 
> * Nativefied:
> 
> ```shell
> docker run -v "${PWD}":/target nativefier --name Pushover -p linux --inject /target/darkreader.js --inject /target/dark.js https://client.pushover.net /target/
> ```
> 
> However I get `Uncaught ReferenceError: DarkReader is not defined` in the console. I'm assuming this has something to do with how the injection is wrapping the object? Is there a way around this?

Same here, Darkreader is always undefined. What is the right way of the injection, so it does work?

**Comment by philippdieter** [comment]
As much as I like to have code included locally I went with this solution as the service I want to nativefy is a cloud service anyway:

```
var script = document.createElement('script');
script.setAttribute('src', 'https://cdn.jsdelivr.net/npm/darkreader@4.9.34/darkreader.min.js');
script.setAttribute('type', 'text/javascript');
script.setAttribute('onLoad', 'DarkReader.enable();');
document.getElementsByTagName('head')[0].appendChild(script);
```

On navigation inside the app this sometimes leads to a flashing white screen so I also injected this css file:

```
html {
    filter: brightness(0);
}
html[data-darkreader-scheme="dark"] {
    filter: unset;
}
```

This will blacken the content until darkreader is loaded

**Comment by luvis** [comment]
@philippdieter 
I tried following your post but it doesn't work for me. Any idea why? I created two files, one .js and one .css and inserted the code you provided and --inject :ed them when running nativefier. No errors while generating the app. But when running the app I get the following

```
onReady ERROR Error: ERR_FAILED (-2) loading 'https://myurl.com/'
    at rejectAndCleanup (node:electron/js2c/browser_init:165:7486)
    at Object.stopLoadingListener (node:electron/js2c/browser_init:165:7861)
    at Object.emit (node:events:394:28) {
  errno: -2,
  code: 'ERR_FAILED',
  url: 'https://myurl.com/'
}
```
The site loades fine but not in dark mode.. Help is appreciated!

**Comment by philippdieter** [comment]
I have no idea.
Can you post the command you run nativefier with?
Have you tried if you get the same result with other urls?
Which os are you running?

**Comment by z1lc** [comment]
What I've been using instead as a poor man's dark mode (responds to Mac OS light/dark mode settings, works especially well with [NightOwl](https://nightowl.kramser.xyz/)):

`dark_mode.css`:
```
@media (prefers-color-scheme: dark) {
    html {
        filter: invert(100%);
    }

    html img {
        filter: invert(100%);
    }
}
```

`nativefier 'calendar.google.com' --name 'Google Calendar' --inject ./Documents/drk.css --darwin-dark-mode-support`

**Comment by TheCleric** [comment]
> What I've been using instead as a poor man's dark mode (responds to Mac OS light/dark mode settings, works especially well with [NightOwl](https://nightowl.kramser.xyz/)):

This is pretty neat, though I don't know about the `img` invert as it leads to weird stuff like this: 🤣 

![image](https://user-images.githubusercontent.com/547567/154752096-546d4fc5-8a1e-40f9-9126-5309e0d52bd8.png)

**Comment by 2ndemosthenes** [comment]
On linux, I created a script that runs [http-server](https://www.npmjs.com/package/http-server) in a directory that contains the darkreader.js file (It's for [V2RayA](https://github.com/v2rayA/v2rayA)):
```sh
#!/bin/bash
http-server -p 6969 /home/user/Applications/V2RayA-Dark/resources/darkreader/ &
/home/user/Applications/V2RayA-Dark/V2RayA;
pkill http-server
```
then made a `v2raya.desktop` file that runs this script.

I also injected the script @philippdieter  suggested but replaced the darkreader.js address with `http://127.0.0.1:6969/darkreader.js`

**Comment by SirFancyBacon** [comment]
Wanted to bump this so people know support is still desired
